google/cloud/storage/bucket_metadata.h

```diff
@@ -458,15 +458,38 @@ class BucketMetadata : private internal::CommonMetadata<BucketMetadata> {
   using CommonMetadata::id;
   using CommonMetadata::kind;
 
-  std::size_t label_count() const { return labels_.size(); }
+  //@{
+  /// @name Accessors and modifiers to the `labels`.
   bool has_label(std::string const& key) const {
     return labels_.end() != labels_.find(key);
   }
   std::string const& label(std::string const& key) const {
     return labels_.at(key);
   }
+  /// Delete a label. This is a no-op if the key does not exist.
+  BucketMetadata& delete_label(std::string const& key) {
+    auto i = labels_.find(key);
+    if (i == labels_.end()) {
+      return *this;
+    }
+    labels_.erase(i);
+    return *this;
+  }
+
+  /// Insert or update the label entry.
+  BucketMetadata& upsert_label(std::string key, std::string value) {
+    auto i = labels_.lower_bound(key);
```

Comment:
I'm amazed that there isn't an upsert method for map, and that something like `mymap["new-key"] = "value"` first creates an empty string as a placeholder until it assigns "value" for the mapped value.  O_o